test/integration/runner_task_test.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+require 'kubernetes-deploy/runner_task'
+class RunnerTaskTest < KubernetesDeploy::IntegrationTest
+  def test_works
+    deploy_fixtures("hello-cloud")
+
+    task_runner = KubernetesDeploy::RunnerTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: @namespace,
+    )
+
+    task_runner.run(
+      task_template: 'hello-cloud-template-runner',
+      command: nil, # idk
+      args: ['a']
+    )
+
+    assert_logs_match(/Starting task runner/)
+  end
+
+  def test_substitutes_arguments
+    deploy_fixtures("hello-cloud")
+
+    task_runner = KubernetesDeploy::RunnerTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: @namespace,
+    )
+
+    pod = task_runner.run(
+      task_template: 'hello-cloud-template-runner',
+      command: nil, # idk
+      args: ['a']
+    )
+
+    assert_equal ['a'], pod.spec.containers.first.args
+  end
+
+  def test_missing_namespace
+    assert_raises(KubernetesDeploy::RunnerTask::FatalTaskRunError) do
+      task_runner = KubernetesDeploy::RunnerTask.new(
+        context: KubeclientHelper::MINIKUBE_CONTEXT,
+        namespace: "missing",
+      )
+
+      task_runner.run(
+        task_template: 'hello-cloud-template-runner',
+        command: nil, # idk
+        args: 'a'
+      )
+    end
+  end
+
+  def test_missing_arguments
```

Comment:
Btw it needs a rebase (hopefully I'll have a chance to do that imminently), but [this branch](https://github.com/Shopify/kubernetes-deploy/compare/secret_ejson) has some examples of using a mock kubeclient to test sad paths / edge cases: https://github.com/Shopify/kubernetes-deploy/blob/33d8d2ebd31997e6f62e9e632a939fea344008e5/test/unit/kubernetes-deploy/ejson_secret_provisioner_test.rb